ddtrace/writer.py

```diff
@@ -120,6 +126,13 @@ def _target(self):
         while True:
             traces = self._trace_queue.pop()
             if traces:
+                # Before sending the traces, make them go through the
+                # processing pipeline
+                try:
+                    traces = self._apply_processing_pipeline(traces)
+                except Exception as err:
+                    log.error("error while processing traces:{0}".format(err))
+            if traces:
```

Comment:
I think you can simply put everything inside the same `if traces`, while keeping the `try-except` separated so a failure in the pipeline will not prevent traces from being sent. Generally speaking, this is correct now and it's the best thing we can do to not introduce a possible huge change.